frontend/src/static/js/main.js

```diff
@@ -126,15 +126,15 @@ window.app = new window.Vue({
             window.alert('Either the .zip file is corrupted, or you uploaded a .zip file that is not generated '
                 + 'by RepoSense.');
           })
-          .then(() => this.updateReportView());
+          .then(() => this.updateReportViewAndRenderTabHash());
     },
-    updateReportDir() {
+    initialize() {
       window.REPORT_ZIP = null;
 
       this.users = [];
-      this.updateReportView();
+      this.updateReportViewAndRenderTabHash();
     },
-    updateReportView() {
+    updateReportViewAndRenderTabHash() {
```

Comment:
It seems that it is not necessary to combine `renderTabHash` with `updateReportView`? I think it may be better to separate it.

Reply:
The reason why I combined these two methods is because `renderTabHash` must be executed after the async method `updateReportView`.

Reply:
Can it be:
```
.then(() => this.updateReportView())
.then(() => this.renderTabHash());
```
? 
In the `window.api.loadSummary()`

Reply:
It should also work, right?

Reply:
Yes this also works but then `this.initialize` will also call `updateReportView` and I'm unsure how to make sure that `renderTabHash` is executed after `updateReportView`. That is why I placed `renderTabHash` to be within `updateReportView`.

Reply:
@fzdy1914 I have modified the code to separate `updateReportView` and `renderTabHash`. But I am unsure if this is the best way of doing this.